src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * This class handles the storage of the current list (i.e. the updating and retrieval).
+ *
+ * @author Megan Wee Rui En
+ * @version CS2103T AY21/22 Semester 1
+ */
+
+package duke.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private final ArrayList<String> commandsSaved = new ArrayList<>(); // array of saved lines in the new txt file
+    private final File FILE;
+
+    public Storage(String filePath) {
+        FILE = new File(filePath);
+    }
+
+    /**
+     * Loads the existing file for the todo list if it exists, or creates a file if it does not.
+     *
+     * @return an ArrayList of tasks for the todo list
+     * @throws IOException
+     */
+    public ArrayList<Task> load() throws IOException {
+
+        final ArrayList<Task> tasks = new ArrayList<>();
+
+        // If the file exists, print out the previous data if it exists
+        if (FILE.exists()) {
+
+            if (FILE.length() == 0) {
+                System.out.println("Oops! Looks like you don't have anything saved :(");
+
+            } else {
+                System.out.println("Here's your progress:");
+                Scanner file_sc = new Scanner(FILE);
```

Comment:
Similar to my previous comment, for consistency sake, `file_sc` can be `fileSc`.